hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneFileSystem.java

```diff
@@ -639,30 +666,25 @@ OzoneListingIterator getDeleteIterator()
    * @return true if successfully deletes all required keys, false otherwise
    * @throws IOException
    */
-  private boolean innerDelete(Path f, boolean recursive) throws IOException {
+  private InnerDeleteResult innerDelete(Path f, boolean recursive)
+      throws IOException {
     LOG.trace("delete() path:{} recursive:{}", f, recursive);
     try {
       OzoneListingIterator iterator =
           new DeleteIteratorFactory(f, recursive).getDeleteIterator();
-      return iterator.iterate();
+      boolean success = iterator.iterate();
+      boolean partiallyDeleted = iterator.isPartiallyComplete();
+      return new InnerDeleteResult(success, partiallyDeleted);
     } catch (FileNotFoundException e) {
       if (LOG.isDebugEnabled()) {
         LOG.debug("Couldn't delete {} - does not exist", f);
       }
-      return false;
+      return new InnerDeleteResult(false, false);
     }
   }
 
-
-  /**
-   * {@inheritDoc}
-   *
-   * OFS supports volume and bucket deletion, recursive or non-recursive.
-   * e.g. delete(new Path("/volume1"), true)
-   * But root deletion is explicitly disallowed for safety concerns.
-   */
-  @Override
-  public boolean delete(Path f, boolean recursive) throws IOException {
+  public InnerDeleteResult getDeleteResult(Path f, boolean recursive)
```

Comment:
Better to change the method name as currently it sounds like it just returns delete result. But actually it's deleting and returning the delete response.